client/src/pages/CalendarPage/CalendarPage.js

```diff
@@ -32,23 +36,70 @@ const dailyTaskDateFormatter = (date) => {
 };
 
 const CalendarPage = () => {
+  const [currDate, setCurrDate] = useState(new Date());
+  const [taskData, setTaskData] = useState([]);
+  const [toggleError, setToggleError] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+  const [toggleErrorMessage, setToggleErrorMessage] = useState("");
+  const [loadingErrorMessage, setLoadingErrorMessage] = useState("");
+  const filterSet = ["Not Started", "Ongoing", "Completed"];
+  const [currentFilter, setCurrentFilter] = useState(filterSet[0]);
+
+  const filterTask = (task) => {
+    switch (currentFilter) {
+      case filterSet[0]: // not started
+        return !task.isStarted;
+      case filterSet[1]: // ongoing
+        return task.isStarted && !("taskEndedAt" in task);
+      case filterSet[2]:
+        return task.isStarted && "taskEndedAt" in task;
+    }
+  };
+
+  const getTasks = async () => {
+    setLoadingErrorMessage("");
+    setLoadingError(false);
+    await instance
+      .get(buildDateRangeRoute(currDate, currDate))
+      .then((taskData) => {
+        const filteredTaskData = taskData.data.filter(filterTask);
```

Comment:
General comment: I think this implementation should be good for now. Normally in the backend we should've actually added a query such that you won't have to filter with another function but this will suffice! 🙂